### PR TITLE
Dockerfile & shell files lint

### LINKED ISSUE #50: [feat] : Improve the CI workflow by adding few more checks.
There are few improvement that can be done here. 

1. Docker file linter: https://github.com/hadolint/hadolint 
2. Bash script linter: https://github.com/koalaman/shellcheck

For the given Dockerfile and Scripts available in the repo.

### PR DESCRIPTION
**Pull request by Azanul**
Fixes #50

### CODE REVIEW

**Comment by shubham-cmyk** [comment]
@Azanul can you fix it.

**Comment by Azanul** [comment]
> @Azanul can you fix lint errors.
> 
> And make sure shell lint is working here.

I did fix the errors but seems like I'll have to deal with warnings and style changes too

**Comment by shubham-cmyk** [comment]
@Azanul We could ignore  ` DL3007,DL3018` here.

**Comment by Azanul** [comment]
> DL3007,DL3018

I think I should add a hadolint config file too

**Comment by Azanul** [comment]
> @Azanul We could ignore ` DL3007,DL3018` here.

I think I should add a hadolint config file too

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -9,19 +9,37 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+
+    - uses: hadolint/hadolint-action@v3.1.0
+      name: Lint Dockerfile
+      with:
+        dockerfile: Dockerfile
```
Comment (shubham-cmyk) [inline .github/workflows/docker-image.yml:16]: we could all the ignore here
```suggestion
        dockerfile: Dockerfile
        ignore: DL3007,DL3018
```

**Comment by shubham-cmyk** [comment]
@Azanul remove the hadolint config file

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (shubham-cmyk) [reply .github/workflows/docker-image.yml:16]: @Azanul Add these ignore to all 3 hadolint

**Comment by shubham-cmyk** [comment]
Convert 

```
RUN curl -fL -Lo /tmp/redis-${REDIS_VERSION}.tar.gz ${REDIS_DOWNLOAD_URL}/redis-${REDIS_VERSION}.tar.gz && \
    cd /tmp && \
    tar xvzf redis-${REDIS_VERSION}.tar.gz && \
    cd redis-${REDIS_VERSION} && \
    make && \
    make install BUILD_TLS=yes
```
to 

```
RUN curl -fL -Lo /tmp/redis-${REDIS_VERSION}.tar.gz ${REDIS_DOWNLOAD_URL}/redis-${REDIS_VERSION}.tar.gz && \
    tar xvzf redis-${REDIS_VERSION}.tar.gz

WORKDIR /tmp/redis-${REDIS_VERSION}

RUN make && \
    make install BUILD_TLS=yes
```

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -9,19 +9,38 @@ jobs:
...
 
   build_dockerfile_exporter:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+
+    - uses: hadolint/hadolint-action@v3.1.0
+      name: Lint Dockerfile.exporter
+      with:
+        dockerfile: Dockerfile.exporter
```
Comment (shubham-cmyk) [inline .github/workflows/docker-image.yml:30]: Add ignore here

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -9,19 +9,38 @@ jobs:
...
 
   build_dockerfile_sentinel:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+
+    - uses: hadolint/hadolint-action@v3.1.0
+      name: Lint Dockerfile.sentinel
+      with:
+        dockerfile: Dockerfile.sentinel
```
Comment (shubham-cmyk) [inline .github/workflows/docker-image.yml:43]: Add ignore here

File: Dockerfile [hunk 4]
```diff
@@ -15,21 +15,22 @@ ARG REDIS_VERSION="stable"
 RUN apk add --no-cache su-exec tzdata make curl build-base linux-headers bash openssl-dev
 
 RUN curl -fL -Lo /tmp/redis-${REDIS_VERSION}.tar.gz ${REDIS_DOWNLOAD_URL}/redis-${REDIS_VERSION}.tar.gz && \
-    cd /tmp && \
-    tar xvzf redis-${REDIS_VERSION}.tar.gz && \
-    cd redis-${REDIS_VERSION} && \
-    make && \
+    tar xvzf redis-${REDIS_VERSION}.tar.gz
+
+WORKDIR /tmp/redis-${REDIS_VERSION}
```
Comment (shubham-cmyk) [inline Dockerfile:20]:
```suggestion
WORKDIR /tmp
```

File: Dockerfile [hunk 5]
```diff
@@ -15,21 +15,22 @@ ARG REDIS_VERSION="stable"
...
+
+RUN make && \
```
Comment (shubham-cmyk) [inline Dockerfile:22]:
```suggestion
RUN tar xvzf redis-${REDIS_VERSION}.tar.gz && \
         make && \
```

**Comment by shubham-cmyk** [comment]
@Azanul 

i am not sure whether shell check actually ran or not ? 
How could run complete in 0s 

image

**Comment by Azanul** [comment]
@shubham-cmyk Looks like it's working

**Comment by shubham-cmyk** [comment]
> @shubham-cmyk Looks like it's working
> 
> https://github.com/OT-CONTAINER-KIT/redis/actions/runs/6435805690/job/17477866384

Yup Looks Like it is working

**Comment by Azanul** [comment]
> Yup Looks Like it is working. It is very fast though.

It that skepticism or joy?

**Comment by shubham-cmyk** [comment]
> > Yup Looks Like it is working. It is very fast though.
> 
> It that skepticism or joy?

Joy. 
I was not sure earlier maybe the runner is not able to calculate the time.

**Review by shubham-cmyk — APPROVED** [review]
LGTM!

**Pull request merged**